### PR TITLE
snippets: Add snippet for timing-critical io

### PR DESCRIPTION
**Pull request by dsprenkels**
This PR adds a snippet which illustrates how assembly directives are used for timing-critical io in microcontrollers. I know that Rust currently does not target AVR (yet), but I think that should not matter that much, right?

### CODE REVIEW

File: snippets/dsprenkels_ws2812b_attiny13.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+* Authors: Daan Sprenkels
+* Reason: timing-critical code
+* Source: https://github.com/dsprenkels/ws2812b_attiny13/blob/master/ws2812b_attiny13.c
+* License: [MIT](https://github.com/dsprenkels/ws2812b_attiny13/blob/master/LICENSE)
```
Comment (bjorn3) [inline snippets/dsprenkels_ws2812b_attiny13.md:5]: Please add `* Language: C` like the other PR.

File: snippets/dsprenkels_ws2812b_attiny13.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            
+            : // No outputs
+            : [lo] "r" (lo), [hi] "r" (hi), [green] "r" (green), [red] "r" (red), [blue] "r" (blue)
+            : "r16", "r17"
+    );
+}
+```
+
+[WS2812B]: https://cdn-shop.adafruit.com/datasheets/WS2812B.pdf
+[ATTiny13A]: http://ww1.microchip.com/downloads/en/DeviceDoc/doc8126.pdf
```
Comment (bjorn3) [inline snippets/dsprenkels_ws2812b_attiny13.md]: Please add a trailing newline.

**Review by bjorn3 — APPROVED** [review]

**Comment by dsprenkels** [comment]
Updated the PR following the comments from @bjorn3.

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**